python/ray/serve/tests/test_failure.py

```diff
@@ -254,5 +257,65 @@ def make_blocked_request():
         assert ray.get(blocked_ref) == "hi"
 
 
+def test_replica_actor_died(serve_instance):
+    """Test replica death paired with delayed handle notification.
+
+    If a replica died, but controller hasn't health-checked it and
+    broadcasted it to handle routers yet, the router should be able
+    to remove it from its replica set.
+    """
+
+    counter = Counter.remote(2)
+
+    @serve.deployment
+    class Dummy:
+        def __call__(self):
+            return os.getpid()
+
+        def check_health(self):
+            ray.get(counter.inc.remote())
+
+    h = serve.run(Dummy.options(num_replicas=2, health_check_period_s=1000).bind())
+
+    deployment_details = get_deployment_details("Dummy", _client=serve_instance)
+    replicas = [r["actor_name"] for r in deployment_details["replicas"]]
+
+    # Wait for controller to health check both replicas
+    ray.get(counter.wait.remote())
```

Comment:
Can you add a timeout to this `ray.get` call?

Reply:
Added a 1s timeout